[CMake] Remove AppleWin build

donny-dont opened the pull request:
#### 5149f37a4ba25cd2e56bffe804b55a0370c36c55
<pre>
[CMake] Remove AppleWin build
<a href="https://bugs.webkit.org/show_bug.cgi?id=251895">https://bugs.webkit.org/show_bug.cgi?id=251895</a>

Reviewed by Brent Fulgham.

Remove AppleWin build completely. Merge any WinCairo specific build
definitions into the general definition.

Sets WinCairo as the default Windows port in build-webkit.

* CMakeLists.txt:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/PlatformWin.cmake:
* Source/JavaScriptCore/shell/PlatformWin.cmake:
* Source/WTF/CMakeLists.txt:
* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/PlatformWin.cmake:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PAL/pal/PlatformAppleWin.cmake: Removed.
* Source/WebCore/PAL/pal/PlatformWin.cmake:
* Source/WebCore/PAL/pal/PlatformWinCairo.cmake: Removed.
* Source/WebCore/PlatformAppleWin.cmake: Removed.
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake: Removed.
* Source/WebDriver/CMakeLists.txt:
* Source/WebInspectorUI/CMakeLists.txt:
* Source/WebKitLegacy/CMakeLists.txt:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/WebKitLegacy/win/WebKitQuartzCoreAdditions/CMakeLists.txt: Removed.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/PlatformWin.cmake:
* Source/cmake/OptionsAppleWin.cmake: Removed.
* Source/cmake/OptionsWin.cmake:
* Source/cmake/OptionsWinCairo.cmake:
* Source/cmake/WebKitCommon.cmake:
* Source/cmake/WebKitFindPackage.cmake:
* Source/cmake/target/PAL.cmake: Removed.
* Source/cmake/target/TargetJavaScriptCore.cmake: Removed.
* Source/cmake/target/TargetWTF.cmake: Removed.
* Source/cmake/target/WebCore.cmake: Removed.
* Tools/DumpRenderTree/PlatformWin.cmake:
* Tools/ImageDiff/PlatformWin.cmake:
* Tools/MiniBrowser/win/CMakeLists.txt:
* Tools/Scripts/webkitdirs.pm:
* Tools/TestWebKitAPI/PlatformWin.cmake:
* WebKitLibraries/win/tools/scripts/COPYRIGHT-END-YEAR: Removed.
* WebKitLibraries/win/tools/scripts/VERSION: Removed.
* WebKitLibraries/win/tools/scripts/auto-version.pl: Removed.
* WebKitLibraries/win/tools/scripts/feature-defines.pl: Removed.
* WebKitLibraries/win/tools/scripts/version-stamp.pl: Removed.

Canonical link: <a href="https://commits.webkit.org/260060@main">https://commits.webkit.org/260060@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dfec4b992bd04405b0e0356a6618d01e462c6d70

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106858 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15893 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39650 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116040 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115599 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110760 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17372 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7093 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99043 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112625 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13151 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96162 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40752 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95035 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27813 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/96373 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9046 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29168 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/95836 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/6987 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9622 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6171 "Found 1 new test failure: fast/images/avif-image-decoding.html (failure)") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/30711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15213 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48715 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/104596 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6943 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11151 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/25907 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
This looks fine to me, as long as the WinCairo and GTK builders remain working.

donny-dont committed:
[CMake] Remove AppleWin build
https://bugs.webkit.org/show_bug.cgi?id=251895

Reviewed by Brent Fulgham.

Remove AppleWin build completely. Merge any WinCairo specific build
definitions into the general definition.

Sets WinCairo as the default Windows port in build-webkit.

* CMakeLists.txt:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/PlatformWin.cmake:
* Source/JavaScriptCore/shell/PlatformWin.cmake:
* Source/WTF/CMakeLists.txt:
* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/PlatformWin.cmake:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PAL/pal/PlatformAppleWin.cmake: Removed.
* Source/WebCore/PAL/pal/PlatformWin.cmake:
* Source/WebCore/PAL/pal/PlatformWinCairo.cmake: Removed.
* Source/WebCore/PlatformAppleWin.cmake: Removed.
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake: Removed.
* Source/WebDriver/CMakeLists.txt:
* Source/WebInspectorUI/CMakeLists.txt:
* Source/WebKitLegacy/CMakeLists.txt:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/WebKitLegacy/win/WebKitQuartzCoreAdditions/CMakeLists.txt: Removed.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/PlatformWin.cmake:
* Source/cmake/OptionsAppleWin.cmake: Removed.
* Source/cmake/OptionsWin.cmake:
* Source/cmake/OptionsWinCairo.cmake:
* Source/cmake/WebKitCommon.cmake:
* Source/cmake/WebKitFindPackage.cmake:
* Source/cmake/target/PAL.cmake: Removed.
* Source/cmake/target/TargetJavaScriptCore.cmake: Removed.
* Source/cmake/target/TargetWTF.cmake: Removed.
* Source/cmake/target/WebCore.cmake: Removed.
* Tools/DumpRenderTree/PlatformWin.cmake:
* Tools/ImageDiff/PlatformWin.cmake:
* Tools/MiniBrowser/win/CMakeLists.txt:
* Tools/Scripts/webkitdirs.pm:
* Tools/TestWebKitAPI/PlatformWin.cmake:
* WebKitLibraries/win/tools/scripts/COPYRIGHT-END-YEAR: Removed.
* WebKitLibraries/win/tools/scripts/VERSION: Removed.
* WebKitLibraries/win/tools/scripts/auto-version.pl: Removed.
* WebKitLibraries/win/tools/scripts/feature-defines.pl: Removed.
* WebKitLibraries/win/tools/scripts/version-stamp.pl: Removed.

Canonical link: https://commits.webkit.org/260060@main
